performance/runner/src/calculate.rs

```diff
@@ -1,278 +1,169 @@
+
+    // the baseline we want to target is the latest one available in this branch
+    let mut target_baseline_dir = baseline_dir.clone();
+    target_baseline_dir.push(latest_version.to_string());
+
+    // these are all the metrics for all available baselines from the target version
+    let baselines: Vec<Baseline> = fs::from_json_files::<Baseline>(&target_baseline_dir)?
         .into_iter()
-        .map(|entry| {
-            let ent: DirEntry = entry
-                .or_else(|e| Err(IOError::ReadErr(results_directory.to_path_buf(), Some(e))))
-                .or_else(|e| Err(CalculateError::CalculateIOError(e)))?;
-
-            Ok(ent.path())
-        })
-        .collect::<Result<Vec<PathBuf>, CalculateError>>()?
-        .iter()
-        .filter(|path| {
-            path.extension()
-                .and_then(|ext| ext.to_str())
-                .map_or(false, |ext| ext.ends_with("json"))
-        })
-        .map(|path| {
-            fs::read_to_string(path)
-                .or_else(|e| Err(IOError::BadFileContentsErr(path.clone(), Some(e))))
-                .or_else(|e| Err(CalculateError::CalculateIOError(e)))
-                .and_then(|contents| {
-                    serde_json::from_str::<Measurements>(&contents)
-                        .or_else(|e| Err(CalculateError::BadJSONErr(path.clone(), Some(e))))
-                })
-                .map(|m| (path.clone(), m))
-        })
-        .collect()
-}
-
-// Given a list of filename-measurement pairs, detect any regressions by grouping
-// measurements together by filename.
-fn calculate_regressions(
-    measurements: &[(&PathBuf, &Measurement)],
-) -> Result<Vec<Calculation>, CalculateError> {
-    /*
-        Strategy of this function body:
-        1. [Measurement] -> [MeasurementGroup]
-        2. Sort the MeasurementGroups
-        3. Group the MeasurementGroups by "run"
-        4. Call `calculate` with the two resulting Measurements as input
-    */
-
-    let mut measurement_groups: Vec<MeasurementGroup> = measurements
-        .iter()
-        .map(|(p, m)| {
-            p.file_name()
-                .ok_or_else(|| IOError::MissingFilenameErr(p.to_path_buf()))
-                .and_then(|name| {
-                    name.to_str()
-                        .ok_or_else(|| IOError::FilenameNotUnicodeErr(p.to_path_buf()))
-                })
-                .map(|name| {
-                    let parts: Vec<&str> = name.split("_").collect();
-                    MeasurementGroup {
-                        version: parts[0].to_owned(),
-                        run: parts[1..].join("_"),
-                        measurement: (*m).clone(),
-                    }
-                })
-        })
-        .collect::<Result<Vec<MeasurementGroup>, IOError>>()
-        .or_else(|e| Err(CalculateError::CalculateIOError(e)))?;
+        .map(|(_, x)| x)
+        .collect();
+
+    // check that we have at least one baseline
+    if baselines.is_empty() {
```

Comment:
If this is a new minor version that we just cut a new release branch for, what will happen in this instance?

Reply:
And by what will happen, I mean will we be able to find a baseline?

Reply:
I'll add a comment to this effect, but if we branch off of main, there will be some old baselines in main. This check was mostly useful for initial development so I'll add that it can be modified once it's all merged and running.